src/expression.js

```diff
@@ -57,7 +57,8 @@ var getKeyComputeData = function (key, scope, readOptions) {
 			if (arguments.length) {
 				observeReader.write(result, observeReader.reads(key), newVal);
 			} else {
-				return observeReader.get(result, key);
+				// Convert possibly numeric key to string, because observeReader.get will do a charAt test on it.
+				return observeReader.get(result, key.toString());
```

Comment:
Commonly cited perf tests support you on this.  I'll try to remember to use that pattern in the future.